toolchain/driver/clang_runner.cpp

```diff
@@ -157,4 +399,98 @@ auto ClangRunner::Run(llvm::ArrayRef<llvm::StringRef> args) -> bool {
   return result == 0 && failing_commands.empty();
 }
 
+// TODO: Replace this with a filesystem cache, and proper management of
+// concurrent cache access. There are a bunch of hacks in this function that
+// aren't realistically reliable for larger use, and only designed as in interim
+// state to separate the implementation of the caching behavior.
+auto ClangRunner::MakeTmpDir() -> void {
+  if (!tmp_dir_.empty()) {
+    return;
+  }
+
+  // Make a temporary directory exclusive to this builder.
+  std::filesystem::path tmp_dir;
+  if (char* tmpdir_env = getenv("TEST_TMPDIR"); tmpdir_env != nullptr) {
+    tmp_dir = std::string(tmpdir_env);
+  } else {
+    tmp_dir = std::filesystem::temp_directory_path();
+  }
+
+  // We use the object's address to avoid collisions between two instances,
+  // whether from two processes or two instances in this process. This is a bit
+  // of a hack perfect, but should be fine until we implement caching at which
+  // point we'll need to do something better here.
+  auto salt = reinterpret_cast<uintptr_t>(this);
+  std::string name = llvm::formatv("carbon-runtimes.{0:x}", salt).str();
```

Comment:
Why doesn't this use something like `mkdtemp` or `llvm::sys::fs::createUniqueDirectory`?

Reply:
Given the TODO to remove, okay. I'll note though, my impression of the LLVM API was that it'd approximately be `std::error_code ec = llvm::sys::fs::createUniqueDirectory(tmp_dir / "carbon-runtimes", tmp_dir_);`, replacing lines 367-391. But I'm okay if you're already preparing to work on the TODO to replace this with a filesystem cache.